### ISSUE TITLE
Question: could wallet_unlock not being blocked by scanning transactions?

### ISSUE THREAD

**Issue by hackfisher** [bitshares/bitshares1-core#510]
In other words, could we put scanning transactions in separate task? Then, unlock rpc call can return immediately and do not need the transactions scanning to complete.

Will there be problems if the user do some operations before or during trxs scanning, e.g like broadcast new transactions(we can valid that if necessary, most other are query actions)?

If the answer is yes, there will be risks, then the web wallet probably should do scanning progress before unlock. (In current implementation, unlock request is pending util scanning finish, and will timeout if the scanning is long enough)

BTW, is it a good idea to stop user do any action before blocks are synced, nxt wallet seems work that way.
